src/core/execute.c

```diff
@@ -861,6 +867,9 @@ static int setup_pam(
                 pam_code = PAM_BUF_ERR;
                 goto fail;
         }
+        strv_free(*env);
+        *env = e;
+        e = NULL;
```

Comment:
what's the rationale for moving this up? we generally try to avoid clobbering call-by-ref parameters on failure. Also see CODING_STYLE. Changing the passed in parameter like you are doing here doesn't look right hence.